### PR TITLE
Add changeset map 

### PR DESCRIPTION
**Pull request by kepta**
This does a bunch of things
- Adds changeset map (it currently uses global var render. need a fix, [issue](https://github.com/osmlab/changeset-map/issues/147)
- Adds a [toastr](https://github.com/tomchentw/react-toastr) element for warnings/notifications
- Handles the changeset-map network requests login and caching on redux level and feeds the data directly to changeset-map module.

cc @ajithranka

### CODE REVIEW

File: public/index.html [hunk 1]
```diff
@@ -6,6 +6,8 @@
   <meta name="viewport" content="width=device-width, initial-scale=1">
   <link rel="shortcut icon" href="%PUBLIC_URL%/favicon.ico">
   <link href="https://api.mapbox.com/mapbox-assembly/v0.12.0/assembly.min.css" rel="stylesheet">
+  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/animate.css/3.5.2/animate.min.css">
```
Comment (ajithranka) [inline public/index.html]: Hey, if possible can we pull this in from the component?

File: src/utils/key_bindings.js [hunk 2]
```diff
@@ -0,0 +1 @@
+export function initKeyGobal() {}
```
Comment (ajithranka) [inline src/utils/key_bindings.js]: We should remove this.

File: src/app.js [hunk 3]
```diff
@@ -5,24 +5,65 @@ import {Changeset} from './views/changeset';
...
+import {initKeyGobal} from './utils/key_bindings';
+import {Navbar} from './components/navbar';
+var ToastMessageFactory = React.createFactory(ToastMessage.animation);
 
 class App extends Component {
+  componentDidMount() {
+    if (document && document.body) {
+      document.body.addEventListener('showToast', this.showToast);
+    }
+    initKeyGobal();
```
Comment (ajithranka) [inline src/app.js]: We should remove this.

File: src/app.js [hunk 4]
```diff
@@ -5,24 +5,65 @@ import {Changeset} from './views/changeset';
...
+        >
+          <ChangesetsList />
+        </Sidebar>
+
         <div
-          className="flex-child flex-child--grow bg-darken10 viewport-twothirds viewport-full-ml"
+          className="flex-child flex-child--grow h-full hmax-full viewport-twothirds"
         >
-          <Route exact path="/" component={Changeset} />
+          <Route exact path="/" component={Changeset} customProp="hi" />
```
Comment (ajithranka) [inline src/app.js]: Cleanup: `customProp="hi"`.

File: src/components/changeset.js [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+    return null;
+  }
+  if (!currentChangeset) return null;
+  const properties = currentChangeset.get('properties');
+
+  return (
+    <div>
+      <Header
+        changesetId={changesetId}
+        date={properties.get('date')}
```
Comment (ajithranka) [inline src/components/changeset.js]: Send all properties?

File: src/components/filters.js [hunk 6]
```diff
@@ -0,0 +1,68 @@
+import React from 'react';
+import filters from '../config/filters.json';
+import {Tooltip} from 'react-tippy';
+
+class Filter extends React.PureComponent {
+  showForm = () => {
```
Comment (ajithranka) [inline src/components/filters.js]: Can we add props?

**Comment by kepta** [comment]
@ajithranka you are good to merge this.

**Comment by ajithranka** [comment]
Thanks @kepta!

**Pull request merged**